zio-interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -177,21 +180,23 @@ private class CatsConcurrentEffect[R](rts: Runtime[R])
     implicit def trace: Trace = InteropTracer.newTrace(cb)
 
     effect.SyncIO {
-      rts.unsafeRun {
-        ZIO
-          .acquireReleaseExitWith(ZIO.descriptor)(
-            (descriptor, exit: Exit[Throwable, A]) =>
-              ZIO.succeed {
-                exit match {
-                  case Exit.Failure(cause) if !cause.interruptors.forall(_ == descriptor.id) => ()
-                  case _ =>
-                    effect.IO.suspend(cb(exit.toEither)).unsafeRunAsync(_ => ())
+      Unsafe.unsafeCompat { implicit u =>
+        rts.unsafe.run {
+          ZIO
+            .acquireReleaseExitWith(ZIO.descriptor)(
+              (descriptor, exit: Exit[Throwable, A]) =>
+                ZIO.succeed {
+                  exit match {
+                    case Exit.Failure(cause) if !cause.interruptors.forall(_ == descriptor.id) => ()
+                    case _ =>
+                      effect.IO.suspend(cb(exit.toEither)).unsafeRunAsync(_ => ())
+                  }
                 }
-              }
-          )(_ => fa)
-          .interruptible
-          .forkDaemon
-          .map(_.interrupt.unit)
+            )(_ => fa)
+            .interruptible
+            .forkDaemon
+            .map(_.interrupt.unit)
+        }.getOrThrowFiberFailure()
```

Comment:
not an expert, but i think maybe this should be `.getOrThrow()` since we're limited to E of Throwable?

Reply:
It was `unsafeRun` before, which was defined as:

```scala
  final def unsafeRun[E, A](zio: ZIO[R, E, A])(implicit trace: Trace): A =
    unsafeRunSync(zio).getOrElse(c => throw FiberFailure(c))
```

so this way we keep the same behavior as before